R/functions_netCDF.R

```diff
@@ -118,68 +300,125 @@ create_netCDF_from_raster_with_variables <- function(x, time_bounds,
+  }
 
-  #--- setup of netCDF file
-  var_chunksizes <- c(raster::ncol(x), raster::nrow(x))
   var_start <- c(1, 1)
 
-  if (has_time_central) {
-    var_chunksizes <- c(var_chunksizes, 1L)
+  if (has_T_timeAxis) {
+    var_chunksizes <- c(var_chunksizes, t_chunksize)
     var_start <- c(var_start, 1)
   }
 
-  # define dimensions
+  if (has_Z_verticalAxis) {
+    var_chunksizes <- c(var_chunksizes, z_chunksize)
+    var_start <- c(var_start, 1)
+  }
+
+  # define dimensions ---------------------------------------------------------------------
+
+  #  bound dimension for lat and long
   bnddim <- ncdf4::ncdim_def(name = "bnds", units = "", vals = seq_len(2L),
     create_dimvar = FALSE)
+
+  # lat and long dimension
   xdim <- ncdf4::ncdim_def(name = "lon", longname = "Longitude",
     units = "degrees_east", vals = xvals)
   ydim <- ncdf4::ncdim_def(name = "lat", longname = "Latitude",
     units = "degrees_north", vals = yvals)
-  if (has_time_central) {
-    tdim <- ncdf4::ncdim_def(name = "time", units = "Gregorian_year since 1900",
-      vals = time_central)
+
+  # time dimension
+  if(has_T_timeAxis) {
+    if (has_time_central) {
+      tdim <- ncdf4::ncdim_def(name = time_attributes[['name']],
+                               units = time_attributes[['units']],
+                               calendar = time_attributes[['calendar']],
+                               vals = time_central)
+    } else {
+      tdim <-  ncdf4::ncdim_def(name = time_attributes[['name']],
+                                units = time_attributes[['units']],
+                                calendar = time_attributes[['calendar']],
+                                vals = time_attributes[['vals']])
+    }
+  }
+
+  # vertical dimension
+  if(has_Z_verticalAxis) {
+    zdim <-  ncdf4::ncdim_def(name = vert_names,
+                              units = vert_units,
+                              vals = vert_vals)
   }
 
-  # define variables
+  # define dimensionality of netcdf variables -------------------------------------------------
   var_dims <- list(xdim, ydim)
-  if (has_time_central) {
+
+  if (has_T_timeAxis) {
     var_dims <- c(var_dims, list(tdim))
   }
 
-  var_defs <- lapply(seq_len(nl), function(k)
+  if (has_Z_verticalAxis) {
+    var_dims <- c(var_dims, list(zdim))
+  }
+
+  if(has_T_timeAxis || has_Z_verticalAxis) nn <- 1 else nn <- nl
+
+  var_defs <- lapply(seq_len(nn), function(k)
     ncdf4::ncvar_def(name = var_names[k], units = var_units[k],
       dim = var_dims, chunksizes = var_chunksizes, missval = NAflag,
       longname = var_longnames[k], prec = ncdf4_datatype))
 
+  # CRS defintion ------------------------------------------------------------------------
   crsdef <- ncdf4::ncvar_def(name = "crs", units = "", dim = list(),
     missval = NULL, prec = "integer")
 
-  # define dimension bounds
+  # define dimension bands ---------------------------------------------------------------
```

Comment:
these are not "dimension bands" but "dimension bounds", here and throughout

Reply:
fixed

Reply:
line 528 is not yet fixed

Reply:
R/functions_netCDF.R, line 528: not fixed, change "bands" to "bounds"

Reply:
fixed